src/core/qgsvectorlayer.h

```diff
@@ -646,6 +677,26 @@ class CORE_EXPORT QgsVectorLayer : public QgsMapLayer, public QgsExpressionConte
      */
     const QgsActionManager *actions() const SIP_SKIP { return mActions; }
 
+    /**
+     * Add a QGIS Server WMS Dimension
+     * \param wmsDimInfo QGIS Server WMS Dimension object with, name, field, etc
+     * \returns TRUE if QGIS Server WMS Dimension has been successfully added
+     * \since QGIS 3.10
+    */
+    bool addWmsDimension( const QgsVectorLayer::WmsDimensionInfo &wmsDimInfo );
```

Comment:
Could we make a new class QgsServerLayerProperties and move this property into that? I'm not a fan of cluttering the top level QgsVectorLayer API with all this server specific settings.

Reply:
I have defined `QgsVectorLayerServerProperties`